raft/raft.go

```diff
@@ -278,7 +280,15 @@ func (r *raft) sendAppend(to uint64) {
 			}
 		}
```

Comment:
What about changing `if r.needSnapshot(pr.Next) {` to `if term, ents, ok := r.lastTermAndEntries(pr.Next, r.maxMsgSize); !ok {`? This can eliminate the recursive function call, and easier to read IMO.